File: frontend/src/types/message.ts
export type MessageContentPart = string | { type: string; image_url: string };

export interface File {
  filename: string;
  content_type: string;
  url?: string;
}

export interface Message {
  id: string;
  chat_id: string;
  role: 'user' | 'assistant';
  content: string | MessageContentPart[];
  file?: File;
  created_at: string;
}

File: backend/models/message.py
from datetime import datetime
from typing import Optional, Union, List, Any
from pydantic import BaseModel, Field


class File(BaseModel):
    """Representation of an uploaded file."""

    filename: str
    content_type: str
    url: Optional[str] = None
    data_uri: Optional[str] = None


class Message(BaseModel):
    """Single chat message."""

    id: str
    chat_id: str
    role: str  # 'user' or 'assistant'
    content: Union[str, List[Any]]
    file: Optional[File] = None
    created_at: datetime = Field(default_factory=datetime.utcnow)
